packages/plugin-core/src/KeybindingUtils.ts

```diff
@@ -1,206 +1,330 @@
+    const userKeybindingConfig = readJSONWithCommentsSync(
+      keybindingConfigPath
+    ) as CommentJSONArray<Keybindings>;
+
+    const alreadyResolved: KeybindingConflict[] = [];
+
+    userKeybindingConfig.forEach((keybinding) => {
+      // find remap / disable of conflicting keybindings
+      let command = keybinding.command;
+      if (command.startsWith("-")) {
+        command = command.substring(1);
+      }
+
+      const resolvedConflict = conflicts.find(
+        (conflict) => conflict.commandId === command
+      );
+      if (resolvedConflict) {
+        alreadyResolved.push(resolvedConflict);
+      }
+
+      // find remap / disable of dendron keybindings
+      command = keybinding.command;
+      if (command.startsWith("-")) {
+        command = command.substring(1);
+      }
+      const resolvedDendronConflict = conflicts.find(
+        (conflict) => conflict.conflictsWith === command
+      );
+
+      if (resolvedDendronConflict) {
+        alreadyResolved.push(resolvedDendronConflict);
+      }
+    });
+
+    return _.differenceBy(conflicts, alreadyResolved);
+  }
+
+  static generateKeybindingBlockForCopy(opts: {
+    entry: Keybindings;
+    disable?: boolean;
+  }) {
+    const { entry, disable } = opts;
+    const whenClause = entry.when ? `  "when": "${entry.when}",` : undefined;
+
+    const args = entry.args
+      ? `  "args": ${JSON.stringify(entry.args)},`
+      : undefined;
+
+    const block = [
+      "{",
+      `  "key": "${disable ? entry.key : ""}",`,
+      `  "command": "${disable ? "-" : ""}${entry.command}",`,
+      whenClause,
+      args,
+      "}",
+      "",
+    ]
+      .filter((line) => line !== undefined)
+      .join("\n");
+    return block;
+  }
+
+  static async showKeybindingConflictPreview(opts: {
+    conflicts: KeybindingConflict[];
+  }) {
+    const md = _md();
+    const { conflicts } = opts;
+    const defaultKeybindingJSON =
+      await KeybindingUtils.openDefaultKeybindingFileAndGetJSON({
+        close: false,
+      });
+    if (defaultKeybindingJSON === undefined) {
+      throw new DendronError({
+        message: "Failed reading default keybinding.",
+        severity: ERROR_SEVERITY.MINOR,
+      });
     }
+    await KeybindingUtils.openGlobalKeybindingFileAndGetJSON({
+      close: false,
+    });
+    const keybindingJSONCommandUri = `command:workbench.action.openGlobalKeybindingsFile`;
+    const defaultKeybindingJSONCommandUri = `command:workbench.action.openDefaultKeybindingsFile`;
+    const contents = [
+      "# Known Keybinding Conflicts",
+      "",
+      "The keybindings listed at the bottom are known to have conflicts with default keybindings for Dendron commands.",
+      "",
+      "Neither Dendron nor the extension may function properly if the keybinding conflict is not resolved.",
+      "",
+      "Consider resolving the keybinding conflicts throught the following methods:",
+      "",
+      "#### 1. Disable conflicting keybindings",
```

Comment:
can we also add a link to a loom video demonstrating this?

Reply:
Link to [Video Guide](https://www.loom.com/embed/82d3fbccd126446bac7a4d16027c07aa) added